File: modules/core/app/model/tools/ToolState.java
package model.tools;

import model.user.Role;
import play.twirl.api.Html;

public enum ToolState {
  
  // @formatter:off
  LIVE("Verfügbare Tools", Role.USER),
  ALPHA("Tools in Alpha-Status", Role.ADMIN),
  BETA("Tools in Beta-Status",Role.ADMIN);
  // @formatter:on
  
  private String description;
  private Role requiredRole;
  
  private ToolState(String theDescription, Role theRequiredRole) {
    description = theDescription;
    requiredRole = theRequiredRole;
  }
  
  public Html getBadge() {
    if(this == LIVE)
      return new Html("");
    else
      return new Html("<sup>" + name() + "</sup>");
  }
  
  public String getDescription() {
    return description;
  }
  
  public Role getRequiredRole() {
    return requiredRole;
  }
  
}

File: app/model/toolMains/ToolState.scala
package model.toolMains

import enumeratum.{EnumEntry, PlayEnum}
import model.Role
import model.Role.RoleUser
import play.twirl.api.Html

import scala.collection.immutable

sealed abstract class ToolState(val german: String, val greek: String, requiredRole: Role) extends EnumEntry {

  def badge: Html = Html(s"<sup>$greek</sup>")

}

object ToolState extends PlayEnum[ToolState] {

  override val values: immutable.IndexedSeq[ToolState] = findValues

  case object LIVE extends ToolState("Verfügbare Tools", "", RoleUser) {
    override def badge: Html = new Html("")
  }

  case object ALPHA extends ToolState("Tools in Alpha-Status", "&alpha;", Role.RoleAdmin)

  case object BETA extends ToolState("Tools in Beta-Status", "&beta;", Role.RoleAdmin)

}
